src/preciceMap.cpp

```diff
@@ -143,22 +143,31 @@ int main(int argc, char *argv[])
   auto              options     = getOptions(argc, argv);
   const std::string meshname    = options["mesh"].as<std::string>();
   const std::string participant = options["participant"].as<std::string>();
+  const std::string dataname    = options["data"].as<std::string>();
+  const bool        isVector    = options["vector"].as<bool>();
 
   auto meshes = aste::BaseName(meshname).findAll(context);
   if (meshes.empty()) {
     throw std::invalid_argument("ERROR: Could not find meshes for name: " + meshname);
   }
 
+  for (auto &mesh : meshes) {
+    mesh.setDataname(dataname);
+  }
+
```

Comment:
We should simply pass the dataname to `Mesh::load()` alongside the dimensions. This simplifies the code overall.